### PR TITLE
DOCS: Added support for arch s390x for Ubuntu, RHEL and SLES

### PR DESCRIPTION
**Pull request by nirmannarang**
Signed-off-by: Nirman Narang <narang@us.ibm.com>

### Proposed changes
Added .md files for SLES and RHEL engine installation.
Added indexing to _data/toc.yaml and engine/install/index.md.
Modified engine/install/index.md, includes/install-script.md, engine/security/rootless.md, storage/storagedriver/device-mapper-driver.md, and storage/storagedriver/select-storage-driver.md to add info for added RHEL and SLES support.
Modified engine/install/ubuntu.md to add s390x repos and other info.
Added tab target for RHEL and SLES to engine/security/rootless.md along with other info.

### CODE REVIEW

File: engine/install/rhel.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
+---
+description: Instructions for installing Docker Engine on RHEL
+keywords: requirements, apt, installation, rhel, rpm, install, uninstall, upgrade, update
```
Comment (thaJeztah) [inline engine/install/rhel.md:3]: Perhaps add `s390x` and `ibm-z` here (I think that makes it pop-up in the auto-complete when searching)

```suggestion
keywords: requirements, apt, installation, rhel, rpm, install, uninstall, upgrade, update, s390x, ibm-z
```

File: engine/install/sles.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
+---
+description: Instructions for installing Docker Engine on SLES
+keywords: requirements, apt, installation, centos, rpm, sles, install, uninstall, upgrade, update
```
Comment (thaJeztah) [inline engine/install/sles.md:3]: Same here

```suggestion
keywords: requirements, apt, installation, centos, rpm, sles, install, uninstall, upgrade, update, s390x, ibm-z
```

File: engine/install/sles.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+title: Install Docker Engine on SLES
+toc_max: 4
+---
+
+To get started with Docker Engine on SLES, make sure you
+[meet the prerequisites](#prerequisites), then
+[install Docker](#installation-methods).
+
+## Prerequisites
+
```
Comment (thaJeztah) [inline engine/install/sles.md:26]: Given that this distro is only available on s390x currently, perhaps it's good to highlight that a bit more, e.g.:

```suggestion
## Prerequisites

> **Note**
>
> We currently only provide packages for SLES on s390x (IBM Z). Other architectures
> are not yet supported for SLES.

```

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks! I left some comments/suggestions inline

**Comment by thaJeztah** [comment]
I pushed https://github.com/docker/docker.github.io/pull/13288 with my suggestions

**Comment by nirmannarang** [comment]
> I pushed #13288 with my suggestions

Thanks, @thaJeztah. For `SLES 15`, the kernel version available is `5.3.18-24.46`, but rootless could not be started with `overlay2`.

**Comment by thaJeztah** [comment]
> Thanks, @thaJeztah. For `SLES 15`, the kernel version available is `5.3.18-24.46`, but rootless could not be started with `overlay2`.

Ah, right, so `5.3` < `5.11`, so from that perspective it should already be covered by the "requires kernel 5.11 or up"

**Pull request closed without merging**